cli.py

```diff
@@ -0,0 +1,72 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import importlib
+import logging
+import os
+import platform
+import multiprocessing
+import click
+
+from click.types import Choice
+from click_default_group import DefaultGroup
+from colorlog.colorlog import ColoredFormatter
+from logging import getLogger
+from multiprocessing import Process
+
+from d3a_interface.exceptions import D3AException
+
+from d3a_api_client.utils import available_client_setups
+
+log = getLogger(__name__)
+
+
+@click.group(name='d3a-api-client', cls=DefaultGroup, default='run', default_if_no_args=True,
+             context_settings={'max_content_width': 120})
+@click.option('-l', '--log-level', type=Choice(list(logging._nameToLevel.keys())), default='INFO',
+              show_default=True, help="Log level")
+def main(log_level):
+    handler = logging.StreamHandler()
+    handler.setLevel(log_level)
+    handler.setFormatter(
+        ColoredFormatter(
+            "%(log_color)s%(asctime)s.%(msecs)03d %(levelname)-8s (%(lineno)4d) %(name)-30s: "
+            "%(message)s%(reset)s",
+            datefmt="%H:%M:%S"
+        )
+    )
+    root_logger = logging.getLogger()
+    root_logger.setLevel(log_level)
+    root_logger.addHandler(handler)
+
+
+_setup_modules = available_client_setups
+
+
+@main.command()
+@click.option('--setup', 'module_name', default="auto_offer_bid_on_device",
+              help="Setup module of client script. Available modules: [{}]".format(
+                  ', '.join(_setup_modules)))
+def run(module_name, **kwargs):
```

Comment:
Clients scripts that are scripts that are inside this repo are working via `d3a-api-client run --setup auto_offer_bid_on_device` thanks for that but can you guide me how to run the client scripts that are outside of this repo?

Reply:
@faizan2590 this was in response to [Spyros comment](https://github.com/gridsingularity/d3a-api-client/pull/82#discussion_r564336361), I can surely add this functionality if it is intended @spyrostz

Reply:
@faizan2590 @Hassan754 I personally for sure wanted the ability to import it as a Python package, because the research team is accustomed to use it like this (because they are used to the d3a behavior). In order to support client scripts that are not part of this repo, this can be done using an extra command line argument (like --setup-file-path) that will navigate to the path of the setup file. Not a fan of this method, due to the fact that one cannot present a list of potential setup files (like the other method is able to), however @faizan2590 if you insist in order to add flexibility to the user then let's add it with an additional argument.

Reply:
![Screen Shot 2021-01-27 at 13 35 20](https://user-images.githubusercontent.com/33482427/105991826-8fe53b80-60a4-11eb-9a4c-0daad738045a.png)
What I meant was that just like `d3a` user can set the base path where all its respective client script could be found and can be imported by `d3a-api-client` like a python module

Reply:
Got it now, and I agree, this is indeed helpful